SSA/Projects/RISCV64/Semantics2.lean

```diff
@@ -0,0 +1,642 @@
+/- we import these specific mathlib libraries to use Nat.cast_ofNat and neg_neg lemmas. -/
+import Mathlib.Order.Lattice
+import Mathlib.Data.Nat.Cast.Defs
+
+/-!
+  ## Dialect semantics bit-vectorization
+This file contains the `RV64` instruction semantics in a form that does
+not contain BitVec.toInt and BitVec.toNat. This file should replace
+the Semantics file in RISCV64 when complete.
+ToInt/ToNat calls hinder proof automation by bv_decide.
+To guarantee faithfullness to the semantics defined by the Sail-Lean RiscV64 model,
+we provide a proof with each rewrite that shows, that the toInt/ toNat free
+version is semantically equivalent to the defintion extracted from
+the Sail RiscV 64 modell in Lean.
+
+
+The instruction semantics suffixed by "bv" indicates that it is
+the purely bit vector defintion where we removed
+toInt/toNat calls.
+-/
+
+/-! ## Helper Bit Vector Lemmas  -/
+/- The lemmas below are helper lemmas used within proofs throughout the file. -/
+
+theorem sshiftRight_eq_sshiftRight_extractLsb {w : Nat}
+    {lw : Nat} (hll : 2^lw = w) (hlw : lw > 0 )
+    (x y : BitVec w)(h : BitVec.toNat y < 2^lw ) :
+        x.sshiftRight y.toNat = x.sshiftRight (y.extractLsb (lw - 1) 0).toNat := by
+  apply BitVec.eq_of_toNat_eq
+  simp
```

Comment:
```suggestion
  simp
```
For this we follow lean's strategy too: having `simp` in the middle of proofs could be problematic, in fact the theorem that the tactic considers are constantly changing, and if we use it like this it might be that in subsequent updates to lean we rely on this code will break because of changes to the `simp` tactic. Therefore, we need to be sure that we know _which_ theorems `simp` actually uses and replace `simp` with `simp only`. 
The only exception to this is when the tactic is at the _end_ of the theorem (terminal simp) or followed by `omega` only, in which case there is no need to use `simp only`.
To find the theorems that are used, you can type `simp?` and apply that suggestion. 
Could you replace all the non-terminal `simp` with `simp only`?